### DISCUSSION TITLE
Cutting Supply of Existing Contract

### DISCUSSION THREAD

**Discussion by rovendoug** [TENK-DAO/tenk#91] [Q&A] [answered]
Hello all!

We are currently in the process of a mint, but are discussing cutting the supply.

Aside from the obvious of closing the NFT sales with a command, is there a way to set a hard ceiling of minted NFTs in the contract, so one doesn't have to baby the contract until the amount desired is sold?

Thanks!

-Rovendoug

**Comment by willemneal** [comment, marked as answer]
Currently, it is not a feature.  Feel free to submit a PR if you are interested in adding it.  The way that I have done it in the past is edit the `tokens_left` function by subtracting some amount and then re-deploying.  Clearly this isn't the best way, but I had been requested infrequently enough that I never added it to the contract itself.

Similar to `nft_burn`.  If you look in branches like `release/x`, you can see examples of that.  Another request I've gotten is to just draw from the raffle but not mint anything.  This will have the same effect as burning since no one will be able to mint that token.
